doc/source/programs/gdal_raster_mosaic.rst

```diff
@@ -1,69 +1,67 @@
-      --hidenodata                                        Makes the VRT band not report the NoData.
-      --addalpha                                          Adds an alpha mask band to the VRT when the source raster have none.
+      -f, --of, --format, --output-format <OUTPUT-FORMAT>  Output format
+      --co, --creation-option <KEY>=<VALUE>                Creation option [may be repeated]
+      -b, --band <BAND>                                    Specify input band(s) number. [may be repeated]
+      --overwrite                                          Whether overwriting existing output is allowed
+      --resolution <xres>,<yres>|average|highest|lowest>   Target resolution (in destination CRS units)
+      --bbox <BBOX>                                        Target bounding box as xmin,ymin,xmax,ymax (in destination CRS units)
+      --target-aligned-pixels                              Round target extent to target resolution
+      --srcnodata <SRCNODATA>                              Set nodata values for input bands. [1.. values]
+      --dstnodata <VRTNODATA>                              Set nodata values at the destination band level. [1.. values]
+      --hidenodata                                         Makes the destination band not report the NoData.
+      --addalpha                                           Adds an alpha mask band to the destination when the source raster have none.
+
 
 
 Description
 -----------
 
-This program builds a :ref:`VRT (Virtual Dataset) <raster.vrt>` that is a mosaic of a list of
-input GDAL datasets. The list of input GDAL datasets can be specified at the end
+This program builds a mosaic of a list of input GDAL datasets, that can be
+either a virtual mosaic in the :ref:`VRT (Virtual Dataset) <raster.vrt>` format,
+or in a more conventional raster format such as GeoTIFF.
+
+The list of input GDAL datasets can be specified at the end
 of the command line or put in a text file (one filename per line) for very long lists.
 Wildcards '*', '?' or '['] of :cpp:func:`VSIGlob` can be used, even on files located
 on network file systems such as /vsis3/, /vsigs/, /vsiaz/, etc.
 
-With :option:`--separate`, each input goes into a separate band in the VRT dataset. Otherwise,
-the files are considered as source rasters of a larger mosaic and the VRT file has the same number of
-bands as the input files.
-
-If one GDAL dataset is made of several subdatasets and has 0 raster bands,
-all the subdatasets will be added to the VRT rather than the dataset itself.
-
-:program:`gdal raster buildvrt` does some checks to ensure that all files that will be put
-in the resulting VRT have similar characteristics: number of bands, projection, color
+:program:`gdal raster mosaic` does some checks to ensure that all files that will be put
+in the resulting file have similar characteristics: number of bands, projection, color
 interpretation, etc. If not, files that do not match the common characteristics will be skipped.
```

Comment:
Outside scope of this PR, but I would suggest a fatal error rather than skipping in this case.

Reply:
captured in ticket https://github.com/OSGeo/gdal/issues/11717